test/extended/images/mongodb_ephemeral.go

```diff
@@ -0,0 +1,59 @@
+package images
+
+import (
+	"fmt"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	"github.com/openshift/origin/test/extended/util/db"
+	"time"
+)
+
+var _ = g.Describe("images: mongodb: ephemeral template", func() {
+	defer g.GinkgoRecover()
+
+	templatePath := exutil.FixturePath("..", "..", "examples", "db-templates", "mongodb-ephemeral-template.json")
+	oc := exutil.NewCLI("mongodb-create", exutil.KubeConfigPath()).Verbose()
+
+	g.Describe("Creating from a template", func() {
+		g.It(fmt.Sprintf("should process and create the %q template", templatePath), func() {
+
+			g.By("creating new app")
+			o.Expect(oc.Run("new-app").Args("-f", templatePath).Execute()).Should(o.Succeed())
+
+			g.By("expecting the mongodb service get endpoints")
+			o.Expect(oc.KubeFramework().WaitForAnEndpoint("mongodb")).Should(o.Succeed())
+
+			g.By("expecting the mongodb service answers for ping")
+
+			podNames, err := exutil.WaitForPods(
+				oc.KubeREST().Pods(oc.Namespace()),
+				exutil.ParseLabelsOrDie("name=mongodb"),
+				exutil.CheckPodIsRunningFn,
+				1,
+				1*time.Minute,
+			)
+			o.Expect(err).ShouldNot(o.HaveOccurred())
+			o.Expect(podNames).Should(o.HaveLen(1))
+
+			mongoPod := podNames[0]
+			mongo := db.NewMongoDB(mongoPod, "")
+			ok, err := mongo.IsReady(oc)
+			o.Expect(err).ShouldNot(o.HaveOccurred())
+			o.Expect(ok).Should(o.BeTrue())
+
+			g.By("expecting that we can insert new record")
+			insertCmd := "db.foo.save({ \"status\": \"passed\" })"
+			result, _ := mongo.Query(oc, insertCmd)
+			o.Expect(result).Should(o.Equal("WriteResult({ \"nInserted\" : 1 })"))
+
+			g.By("expecting that we can read record")
+			findCmd := "db.foo.find({}, {_id: 0})" // don't include _id field to output because it changes every time
+			result, _ = mongo.Query(oc, findCmd)
+			o.Expect(result).Should(o.Equal("{ \"status\" : \"passed\" }"))
```

Comment:
use `ContainSubstring` instead of Equal( ... that will explode as soon as they add some new line or whitespace into output. (same for the statement above)